### PR TITLE
Fix Clawpatch review findings

### PR DESCRIPTION
**Pull request by chapati23**
## Summary
- Fix dropped-swap leaderboard heartbeat flushing so zero-USD or missing-caller swaps still close stale leaderboard windows.
- Add Arkham refresh-rotation coverage and Oracle hover text handling for non-finite deviation samples.
- Replace dead health-badge buttons with accessible static details and wire InfoPopover ids/descriptions.
- Exclude generated `.clawpatch/**/*.json` state from Trunk formatting and record follow-up BACKLOG items for each scoped max-lines ESLint bypass.

## Invariants
- A swap event with a valid chain/timestamp/block still advances the leaderboard heartbeat even when trader rollups are intentionally skipped.
- Arkham refresh mode rotates existing Arkham-sourced labels oldest-first before applying `limit`.
- Untrusted Oracle health samples never render misleading `NaN%` deviation hover text.
- Non-interactive health badges are not exposed as dead buttons.

## Degraded behavior
- Uncomputable leaderboard volume remains excluded from trader/pool/aggregator rollups, while the chain heartbeat still flushes closed-day snapshots.
- Oracle chart hover text falls back to `Deviation: N/A` when deviation is non-finite.
- `.clawpatch` JSON remains tracked as generated review state, but Trunk no longer formats it.

## Intentional non-goals
- Splitting existing oversized dashboard/indexer functions is deferred to the BACKLOG entries added in this PR.
- No dashboard query or SWR polling behavior is changed.

## Test plan
- `pnpm agent:quality-gate --run`
- Browser verification with local Next dev server against public Hasura: Oracle tab rendered, no alerts, no console errors, and no `NaN` in Plotly hover text.
- `clawpatch status --plain` reports `openFindings: 0`.

---

> [!NOTE]
> **Medium Risk**
> Touches indexer leaderboard rollup/flush behavior, which can affect snapshot correctness and downstream dashboard data, though changes are narrow and covered by new tests. UI changes are mostly presentation/a11y with low blast radius.
> 
> **Overview**
> Fixes an indexer edge case where `applyLeaderboardSnapshots` would early-return on missing caller or `volumeUsdWei === 0n` *without* running the leaderboard window heartbeat, by invoking the flush path even when trader rollups are intentionally skipped and adding regression coverage for closed-day snapshot advancement.
> 
> Improves dashboard robustness and accessibility: Oracle chart hover text now renders `N/A` for non-finite price/deviation via a shared formatter (with unit tests), Arkham enrichment gains a test pinning refresh rotation (oldest `updatedAt` first before `limit`), health badges are no longer rendered as inert buttons (static details + screen-reader text), and `InfoPopover` now wires `aria-controls`/`aria-describedby` to a stable tooltip id.
> 
> Housekeeping: Trunk ignores Prettier formatting for generated `.clawpatch/**/*.json`, ESLint baselines are updated for line shifts/scoped disables, and `BACKLOG.md` records follow-ups to split oversized functions introduced by these scoped fixes.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit cf614f3d678a6353712e9f4856e21ab84d989c01. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by chapati23** [comment]
@claude review

**Comment by chapati23** [comment]
@claude review feedback addressed in a6ae5e3d:\n\n- Fixed: added regression coverage for the missing-caller heartbeat path in indexer-envio/test/leaderboardSnapshots.test.ts.\n- Fixed: guarded non-finite Oracle hover prices with Price: N/A and added a unit test.\n- Fixed: clarified that the Arkham refresh-rotation test intentionally covers both legacy ARKHAM_TAG and source: "arkham" attribution formats.\n- Note: the closure capture in flushLeaderboardWindow is intentional and unchanged.

File: .trunk/trunk.yaml [hunk 1]
```diff
@@ -23,6 +23,9 @@ lint:
         - "**/coverage/**"
         - "**/generated/**"
         - "**/.envio/**"
+        # Clawpatch owns these generated state artifacts; formatting them
+        # creates churn without improving source review.
+        - .clawpatch/**/*.json
```
[the remark this answers is not in the document]
Comment (chapati23) [reply .trunk/trunk.yaml:30]: Fixed in cf614f3d — narrowed the generated `.clawpatch/**/*.json` Trunk ignore to `prettier` only, so TruffleHog and the rest of Trunk still scan Clawpatch artifacts while formatting churn is avoided.
